src/apps/companies/apps/add-company/client/AddCompanyForm.jsx

```diff
@@ -66,14 +66,21 @@ function AddCompanyForm({
     return {}
   }
 
+  function transformPayload(values) {
+    return {
+      ...omit(values, [!values?.telephone_number && 'telephone_number']),
```

Comment:
This took me a moment to understand what it's doing. The use of && to passthru a truthy string was astonishing to me, (even though || to do the same in Ruby is kinda normal), combined with a not in there. I got there in the end, but I do wonder whether this could be written in a way that would be more easily understood by a mixed seniority team.

That said I'm not blocking this PR.

```suggestion
      ...omit(values, values?.telephone_number ? [] : ['telephone_number']),
```

Reply:
i used array spreading primarily because it seems to be quite common throughout this code repository. kids this days spread their arrays like nothing.

Reply:
scary